File: leetcode/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ans = ""
        n = len(word1)
        m = len(word2)
        for i in range(min(len(word1), len(word2))):
            ans += (word1[i] + word2[i])

        k = min(len(word1), len(word2))

        while n > min(len(word1), len(word2)):
            ans += word1[k]
            k += 1
            n -= 1
        
        while m > min(len(word1), len(word2)):
            ans += word2[k]
            k += 1
            m -= 1
        return ans

File: leetcode/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    ans = []
    for (let i = 0; i < Math.min(word1.length, word2.length); i++) {
        ans.push(word1[i])
        ans.push(word2[i])
    }
    if (word1.length <= word2.length) {
        r = word1.length
        while (r < word2.length) {
            ans.push(word2[r])
            r++;
        }
    } else { 
        r = word2.length
        while (r < word1.length) {
            ans.push(word1[r])
            r++;
        }

    }
  return ans.join('')  
};